app/actions/sendModalActions.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import { createActions } from 'spunky'
+import { noop } from 'lodash-es'
+import parseQRCode, { type RecipientData } from '../util/parseQRCode'
+
+export const ID = 'send'
+
+export const getRecipientData = createActions(
+  ID,
+  (url: string) => (): RecipientData => {
+    try {
+      return parseQRCode(url)
+    } catch (msg) {
+      throw new Error(msg)
+    }
+  }
+)
+
+export const clearRecipientData = createActions(ID, () => () => null)
```

Comment:
Is there a better way to do this?
`getRecipientData` sets `recipientData` and `clearRecipientData` empties it. This function `() => () => null` just looks a bit weird and unnecessary.. Is there a `spunky` way to clear the data?

Reply:
Not sure. Maybe @evgenyboxer has some time to look at this. Otherwise, I will try and do some research later today or tomorrow.

Reply:
Yeah, I believe this is exactly what `withReset` is for - 

```
withReset(getRecipientData, ((prevProps, nextProps) => {
  // put your reset condition here
})
```

hopefully that helps! (you can put it after your `withData` higher order component in your compose function.

Reply:
I looked into that before and it didn't make sense to me in this context actually. Lmk if you think otherwise.